gh-129438: Update ``--enable-experimental-jit`` section with install requirements (#129450)
python/cpython: Tools/jit/README.md

This version of CPython can be built with an experimental just-in-time compiler[^pep-744]. While most everything you already know about building and using CPython is unchanged, you will probably need to install a compatible version of LLVM first.

> Note that Python 3.11 or newer is required to build JIT.

Reviewer: ```suggestion
Python 3.11 or newer is required to build the JIT.
```
